### ISSUE TITLE
Add Restart/backup Planer

### ISSUE THREAD

**Issue by Fire041992** [BlueSquare23/web-lgsm#20]
**Is your feature request related to a problem? Please describe.**
Currently, the lack of an automated Restart/Backup planner requires server administrators to manually restart the server and create backups. This is especially problematic for maintaining server performance and ensuring data security. Without regular restarts, the server can become unstable, leading to potential crashes that could disrupt gameplay. Additionally, without frequent backups, there is a significant risk of data loss, particularly in the event of an unexpected server failure. Automating these tasks would greatly enhance the efficiency of server management, reduce the likelihood of server issues, and protect valuable game data."

**Describe the solution you'd like**
The solution I envision is a fully automated Restart and Backup Planner integrated into the web interface. This feature would allow administrators to schedule server restarts at specific times, with customizable intervals (daily, weekly, or custom). It should include automated notifications to players before each restart and a graceful shutdown option to ensure that gameplay is not abruptly interrupted.

**Describe alternatives you've considered**
"In considering alternatives, I looked at manual management, which offers control but is time-consuming and prone to human error. Another option was using server-side scripts with Cron jobs or Task Scheduler, providing flexibility but lacking integration with the web interface, making it harder to manage. I also considered external backup and monitoring services, which are convenient but introduce reliance on third parties and potential costs.

**Comment by BlueSquare23** [comment]
Yeah not a bad idea. I can definitely see how having control of that from within the web interface would be handy, even just for scheduling regular restarts like you mentioned. I'll probably just make a simple interface to wrap cron. But yeah should certainly be doable! 

I'm currently working on some pretty big changes to this project on the dev branch with a lot of things in the backend currently in-flux. See, [dev-1.8 Todos.txt](https://github.com/BlueSquare23/web-lgsm/blob/a61bf5a2d2a4da6bc3840a0da44f88b808436581/Todos.txt) if you're curious. I want to get those things done first before starting on this new feature because A) they'll help with the development of this feature and B) this will just make more sense to add after those things are in place.

All that to say, I don't know if this will make it into the v1.8.0 release, but def make it into 1.9! But yeah thanks for bringing it up! I'm going to put it in the Todo.txt backlog for right now and will update this issue as things progress.

**Comment by BlueSquare23** [comment]
Still a long way from being done. None of the backend code for this has been figured out yet. But I got a pretty looking modal with a form in it and some basic javascript setup. Its a start! Anyways, yeah know its been over a year, but figured I'd update you. Am finally working on this, and will 🤞 get it out with the release of v1.8.5. Will update you then!

![Image](https://github.com/user-attachments/assets/ccd95a62-e2fa-4338-a861-25f7d2bb1e9b)

**Comment by BlueSquare23** [comment]
It has been a long time coming and thank you for your patience. Happy to announce, with the release of [v1.8.5](https://github.com/BlueSquare23/web-lgsm/releases/tag/v1.8.5) there is now a builtin jobs editor in the web interface!

By default, its set to only allow game server commands as cronjobs (aka stop, start, restart, update, etc). However, you can also now set `allow_custom_jobs` flag in the main.conf to allow the creation of any arbitrary cronjob under the sun. Although of course be advised that has security implications, why its disabled by default.

So say for example you wanted to have regularly scheduled game server restarts. You could make one job with a _"send"_ command that says "game server restarting in 15 minutes." And then make a second job, scheduled 15 minutes later to actually do the restart.

Thanks again for the suggestion! Sorry took me so long to get to it, but its out there now!
